src/libMessage/ZilliqaMessage.proto

```diff
@@ -310,15 +310,15 @@ message ProtoAccountStore
 message ProtoPeer
 {
     ByteArray ipaddress    = 1;
-    uint32 listenporthost  = 2;
+    oneof oneof2 { uint32 listenporthost  = 2; }
 }
 
 message ProtoPoWSolution
 {
     uint64 nonce        = 1;
```

Comment:
Any reason, we don't apply for `oneof` to `nonce` field?

Reply:
> Any reason, we don't apply for `oneof` to `nonce` field?

I think a zero nonce for pow solution is highly unlikely?

Reply:
In general, what is the reason for skipping other fields like `version` and `gaslimit`. These were optional in proto2 version previously.

Reply:
so issue comes when the default value of these are actual values in the block (in this case the default value is 0). version starts from one and gaslimit is also non-zero.